include/powsybl/iidm/ShuntCompensatorLinearModelAdder.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_SHUNTCOMPENSATORLINEARMODELADDER_HPP
+#define POWSYBL_IIDM_SHUNTCOMPENSATORLINEARMODELADDER_HPP
+
+#include <powsybl/iidm/ShuntCompensatorModelAdder.hpp>
+#include <powsybl/stdcxx/math.hpp>
+#include <powsybl/stdcxx/optional.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+class ShuntCompensatorAdder;
+
+namespace shunt_compensator {
+
+class ShuntCompensatorLinearModelAdder : public ShuntCompensatorModelAdder {
+public:  // ShuntCompensatorModelAdder
+    ShuntCompensatorAdder& add() override;
+
+public:
+    explicit ShuntCompensatorLinearModelAdder(ShuntCompensatorAdder& parent);
+
+    ShuntCompensatorLinearModelAdder(ShuntCompensatorAdder& parent, const ShuntCompensatorLinearModelAdder& adder);
+
+    ShuntCompensatorLinearModelAdder& setBPerSection(double bPerSection);
+
+    ShuntCompensatorLinearModelAdder& setGPerSection(double gPerSection);
+
+    ShuntCompensatorLinearModelAdder& setMaximumSectionCount(unsigned long maximumSectionCount);
+
+private:  // ShuntCompensatorModelAdder
+    std::unique_ptr<ShuntCompensatorModel> build(ShuntCompensator& shuntCompensator) const override;
+
+    std::unique_ptr<ShuntCompensatorModelAdder> clone(ShuntCompensatorAdder& parent) const override;
+
+    unsigned long getMaximumSectionCount() const override;
+
+private:
+    ShuntCompensatorAdder& m_parent;
```

Comment:
To be discussed: we have to decide if the sub-model adder has to be assignable. In DanglingLineGenerationAdder and HalfLineAdder we use a `std::reference_wrapper`. We have to be consistent.